src/org/traccar/protocol/Ardi01ProtocolDecoder.java

```diff
@@ -78,7 +78,7 @@ protected Object decode(
         position.set(Position.KEY_SATELLITES, satellites);
 
         position.set(Position.KEY_EVENT, parser.next());
-        position.set(Position.KEY_BATTERY, parser.next());
+        position.set(Position.KEY_BATTERY_LEVEL, parser.nextInt(0));
```

Comment:
In this case value is not optional, so it's not a problem, but can you please check that it's case for all the changed protocols. If the value is optional, it's better to use `nextInt` without argument, so the value is not included in the attributes instead of including zero.